File: tests/evaluation/metadl-java/src/class-hierarchy-utils/Test1.java
class A {
	Object f() { return null; }

	static Object bar() { return null; }
}

class B extends A {
	B f() { return this; }

	static Object bar() { return null; }
}

interface I {
	int foo();
	int foo(int x);
	int foo(Object x);


}

interface J extends I {
}

class C extends B implements I {
	public int foo() { return 0;}
	public int foo(int x) { return 0;}
	public int foo(Object x) { return 0;}
}

class D<T> {

}

class E<T> extends D<T> implements I {
	public int foo() { return 0;}
	public int foo(int x) { return 0;}
	public int foo(Object x) { return 0;}
}

enum F {
	A {
		@Override
		public int fee() { return 1; }
	}

	;

	public int fee() {
		return 5;
	}
}

class Test1 {
	public static void main(String[] args) {
		//int x = F.A.foo();

	    I i = new I() {
				public int foo() {
					return 0;
				}
				public int foo(int x) { return 0;}
				public int foo(Object x) { return 0;}

			};
	}
}

File: tests/clang/evaluation/src/null_pointer_dereference/test_1.c
#include <stdlib.h>

int foo(int *p, int *q, int n) {
  for (int i = 0; i < n; ++i) {
    *p = 0;
  }
  return 1;
}

/* The checker should be able to detect that p is not NULL when calling foo,
   and it shouldn't produce any warning. */
int bar(int *p, int *q) {
  int cnt = 0;
  if (p != NULL) {
    cnt += foo(p, q, 10);
  }
  return 0;
}

int entry(void) {
  bar (NULL, NULL);
}
